x-pack/plugins/cloud/public/plugin.ts

```diff
@@ -41,6 +42,7 @@ export interface CloudConfigType {
 interface CloudSetupDependencies {
   home?: HomePublicPluginSetup;
   security?: Pick<SecurityPluginSetup, 'authc'>;
+  application?: Promise<ApplicationStart>;
```

Comment:
Can you please introduce a distinct type for the input of `setupFullstory` instead of altering the plugin's setup contract deps? That would also allow to remove the inline `& { basePath: IBasePath }` L176

Reply:
`application?: Promise<ApplicationStart>;` needs to move to `SetupFullstoryDeps`